src/tasks/tasks.py

```diff
@@ -543,6 +546,101 @@ def get_metrics(self, reset=False):
+        self.n_classes = 2
+        self.val_metric = "%s_mcc" % self.name
+        self.val_metric_decreases = False
+        self.scorer1 = Correlation("matthews")
+        self.scorer2 = CategoricalAccuracy()
+        self.scorers = [self.scorer1, self.scorer2]
+    
+    def load_data(self, path, max_seq_len):
+        '''Load the data'''
+        tag_vocab = vocabulary.Vocabulary(counter=None)
+        self.train_data_text = load_tsv(self._tokenizer_name, os.path.join(path, "acceptability_minimal_pairs.tsv"), max_seq_len,
+                           s1_idx=1, s2_idx=2, label_idx=3, tag2idx_dict={'source': 0, 'case': 4}, tag_vocab=tag_vocab)
+        self.val_data_text = self.test_data_text = self.train_data_text
+        # Create score for each tag from tag-index dict
+        self.tag_list = get_tag_list(tag_vocab)
+        self.tag_scorers1 = create_subset_scorers(
+            count=len(
+                self.tag_list),
+            scorer_type=Correlation,
+            corr_type="matthews")
+        # self.tag_scorers2 = create_subset_scorers(
+        #     count=len(self.tag_list), scorer_type=CategoricalAccuracy)
+
+        log.info("\tFinished loading CoLA minimal pairs.")
+        return
+
+    def process_split(self, split, indexers):
+        def _make_instance(input1, input2, labels, tagids):
+            ''' from multiple types in one column create multiple fields
+            input0: shared part (masked form) of both sentence (only used in BERT)
+            input1: sentence1
+            input2: sentence2
+            index: which token is different
+            labels: is 
+            tagmask: which tag the pair has, in this dataset, the source of data 
+            '''
+            d = {}
+            d["input1"] = sentence_to_text_field(input1, indexers)
+            d["sent1_str"] = MetadataField(" ".join(input1[1:-1]))
+            d["input2"] = sentence_to_text_field(input2, indexers)
+            d["sent2_str"] = MetadataField(" ".join(input2[1:-1]))
+            mask_index = [i for i in range(len(input1)) if input1[i] != input2[i]][0]
+            input0 = [i for i in input1]
+            input0[mask_index] = BERT_MASK_TOK
+            d["input0"] = sentence_to_text_field(input0, indexers)
+            d["sent0_str"] = MetadataField(" ".join(input0[1:-1]))
+            d["index"] = IndexField(mask_index, d["input1"])
+            d["labels"] = LabelField(labels, label_namespace="labels",
+                                     skip_indexing=True)
+            d["tagmask"] = MultiLabelField(tagids, label_namespace="tagids",
+                                           skip_indexing=True, num_labels=len(self.tag_list))
+                                           
+            return Instance(d)
+
+        instances = map(_make_instance, *split)
+        return instances  # lazy iterator
+    
+    def update_metrics(self, logits, labels, tagmask=None):
+        logits, labels = logits.detach(), labels.detach()
```

Comment:
do you need to detach explicitly? I thought within scorers we're calling `.item()` or something similar anyways.